torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -1037,6 +1038,7 @@ void ProcessGroupNCCL::workCleanupLoop() {
         } else {
           it = workMetaList_.erase(it);
         }
+        at::cuda::CUDAGraph::dec_pending_event_queries();
```

Comment:
@Aidyn-A this is where we decrement, when the work is actually enqueud and done and cleaned up.